build.rs

```diff
@@ -0,0 +1,21 @@
+fn main() {
+    let ac = autocfg::new();
+    if ac.probe_expression("format!(\"{:e}\", 0_i32)") {
+        if !(ac.probe_expression("format!(\"{:e}\", 0_i8)")
+            && ac.probe_expression("format!(\"{:e}\", 0_i16)")
+            && ac.probe_expression("format!(\"{:e}\", 0_i32)")
+            && ac.probe_expression("format!(\"{:e}\", 0_i64)")
+            && ac.probe_expression("format!(\"{:e}\", 0_i128)")
+            && ac.probe_expression("format!(\"{:e}\", 0_u8)")
+            && ac.probe_expression("format!(\"{:e}\", 0_u16)")
+            && ac.probe_expression("format!(\"{:e}\", 0_u32)")
+            && ac.probe_expression("format!(\"{:e}\", 0_u64)")
+            && ac.probe_expression("format!(\"{:e}\", 0_u128)"))
+        {
+            panic!("Some integer types implement *Exp traits, but not others")
```

Comment:
This is a bit overkill, and has a tangible effect on compile times -- I measured that running the build script on current stable takes ~110ms, whereas nightly takes ~680ms. After all, each probe requires a new `rustc` process invocation.

If you _really_ want to test them all, it could be done better in one large probe expression. However, we really only care about this feature for testing purposes, so I don't think it needs to be so thorough. For that matter, the test only uses it for `Rational = Ratio<isize>`, which you didn't even probe... :wink: 

Let's just probe for support in `isize` and call it good.

Reply:
> each probe requires a new rustc process invocation

I didn't even think about how it worked, but that makes sense.

> Let's just probe for support in isize and call it good.

Sounds good.